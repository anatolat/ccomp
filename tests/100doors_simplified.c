#include <stdio.h>
 
int main()
{
  char is_open[100];
  int pass; int door;

  for (door = 0; door < 100; ++door)
    is_open[door] = 0;
 
  /* do the 100 passes */
  for (pass = 0; pass < 100; ++pass)
    for (door = pass; door < 100; door = door + pass+1)
      is_open[door] = !is_open[door];
 
  /* output the result */
  for (door = 0; door < 100; ++door) {
    char *s;
    if (is_open) {
      s = "open";
    } else {
      s = "closed";
    }
 
    printf("door #%d is %s.\n", door+1, s);
  }
 
  return 0;
}