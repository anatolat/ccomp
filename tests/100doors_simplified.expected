door #1 is open.
door #2 is open.
door #3 is open.
door #4 is open.
door #5 is open.
door #6 is open.
door #7 is open.
door #8 is open.
door #9 is open.
door #10 is open.
door #11 is open.
door #12 is open.
door #13 is open.
door #14 is open.
door #15 is open.
door #16 is open.
door #17 is open.
door #18 is open.
door #19 is open.
door #20 is open.
door #21 is open.
door #22 is open.
door #23 is open.
door #24 is open.
door #25 is open.
door #26 is open.
door #27 is open.
door #28 is open.
door #29 is open.
door #30 is open.
door #31 is open.
door #32 is open.
door #33 is open.
door #34 is open.
door #35 is open.
door #36 is open.
door #37 is open.
door #38 is open.
door #39 is open.
door #40 is open.
door #41 is open.
door #42 is open.
door #43 is open.
door #44 is open.
door #45 is open.
door #46 is open.
door #47 is open.
door #48 is open.
door #49 is open.
door #50 is open.
door #51 is open.
door #52 is open.
door #53 is open.
door #54 is open.
door #55 is open.
door #56 is open.
door #57 is open.
door #58 is open.
door #59 is open.
door #60 is open.
door #61 is open.
door #62 is open.
door #63 is open.
door #64 is open.
door #65 is open.
door #66 is open.
door #67 is open.
door #68 is open.
door #69 is open.
door #70 is open.
door #71 is open.
door #72 is open.
door #73 is open.
door #74 is open.
door #75 is open.
door #76 is open.
door #77 is open.
door #78 is open.
door #79 is open.
door #80 is open.
door #81 is open.
door #82 is open.
door #83 is open.
door #84 is open.
door #85 is open.
door #86 is open.
door #87 is open.
door #88 is open.
door #89 is open.
door #90 is open.
door #91 is open.
door #92 is open.
door #93 is open.
door #94 is open.
door #95 is open.
door #96 is open.
door #97 is open.
door #98 is open.
door #99 is open.
door #100 is open.
